data_deletion/raw_data.py

```diff
@@ -1,31 +1,55 @@
-import operator
 from os import listdir
 from os.path import join, isdir
+import datetime
 from egcg_core import rest_communication
 from egcg_core.constants import ELEMENT_RUN_NAME, ELEMENT_PROCS, ELEMENT_STATUS, DATASET_DELETED, ELEMENT_PROC_ID
 from egcg_core.config import cfg
 from data_deletion import Deleter
 
+reporting_app_date_format = '%d_%m_%Y_%H:%M:%S'
+
 
 class RawDataDeleter(Deleter):
-    def __init__(self, work_dir, dry_run=False, deletion_limit=None, manual_delete=None):
-        super().__init__(work_dir, dry_run, deletion_limit)
-        self.deletable_sub_dirs = ('Data', 'Logs', 'Thumbnail_Images')
-        self.data_dir = cfg['data_deletion']['raw_data']
+    alias = 'raw'
+    deletable_sub_dirs = ('Data', 'Logs', 'Thumbnail_Images')
+
+    def __init__(self, cmd_args):
+        super().__init__(cmd_args)
+        self.raw_data_dir = cfg['data_deletion']['raw_data']
         self.archive_dir = cfg['data_deletion']['raw_archives']
-        self.list_runs = manual_delete
+
+        now = self._now()
+        self.deletion_threshold = now - datetime.timedelta(days=cfg['data_deletion'].get('run_age_threshold_in_days', 14))
 
     def deletable_runs(self):
-        runs = rest_communication.get_documents('aggregate/all_runs', paginate=False, quiet=True, sort=ELEMENT_RUN_NAME)
-        deletable_runs = []
-        for r in runs:
-            if (self.list_runs and r[ELEMENT_RUN_NAME] in self.list_runs) or self._run_deletable(r):
-                deletable_runs.append(r)
-        deletable_runs.sort(key=operator.itemgetter('run_id'))
-        return deletable_runs[:self.deletion_limit]
+        manual_runs = [
+            rest_communication.get_document('runs', where={'run_id': run_id})
+            for run_id in self.manual_delete
+        ]
+        auto_runs = rest_communication.get_documents(
+            'runs',
+            where={
+                'aggregated.review_statuses': {'$ne': 'not reviewed'},
+                'aggregated.most_recent_proc.status': {'$in': ['finished', 'aborted']}
+            }
+        )
+        runs = manual_runs + [r for r in auto_runs if self._run_old_enough_for_deletion(r['run_id'])]
+        return runs[:self.deletion_limit]
+
+    def _run_old_enough_for_deletion(self, run_id):
+        run_elements = rest_communication.get_documents('run_elements', where={'run_id': run_id}, all_pages=True)
```

Comment:
I think we'll need to ignore the `unknown` barcode as it will not get reviewed and won't have a `useable_date`